dev/tools/update_icons.dart

```diff
@@ -262,52 +268,76 @@ Error: New codepoints file does not contain all the existing codepoints.\n
   }
 }
 
-String _generateDeclaration(MapEntry<String, String> tokenPair) {
-  final String description = tokenPair.key.replaceAll('_', ' ');
-
-  String styleSuffix = '';
-  String webFontKey = tokenPair.key;
-
-  // The first line of each generated declaration includes a comment of html.
-  // DartDocs reads that to make the listings in our api docs that shows the
-  // icon rendered next to its key name. Unfortunately, unlike Flutter, this
-  // html needs to use a different web font for each style. We read the style's
-  // suffix from the key for Flutter's icons font, add the corresponding style's
-  // suffix to the class we pass into html, and then remove the suffix from the
-  // icon key. The keys needed for the individual web fonts do not use a suffix
-  // to denote style.
-  if (webFontKey.endsWith('_outlined') && webFontKey!='insert_chart_outlined') {
-    styleSuffix = '-outlined';
-    webFontKey = webFontKey.replaceAll('_outlined', '');
-  }
-  if (webFontKey.endsWith('_rounded')) {
-    styleSuffix = '-round';
-    webFontKey = webFontKey.replaceAll('_rounded', '');
+enum IconStyle {
+  regular,
+  outlined,
+  rounded,
+  sharp,
+}
+
+extension IconStyleSuffix on IconStyle {
+  // The suffix for the 'material-icons' HTML class
+  String suffix() {
+    switch (this) {
```

Comment:
this switch statement doesn't have a case for iconstyle.regular

Reply:
My assumption is regular has no prefix that is why it defaulted. I wonder in this case if it should still have its own case and it returning an empty string. I am not too sure about this. It seems like the way you implemented it looks cleaner.

Reply:
I would do it, but you end up having to do either of these to make the analyser happy, neither of which feel right

```
    switch (this) {
      case IconStyle.regular: return '';
      case IconStyle.outlined: return '-outlined';
      case IconStyle.rounded: return '-round';
      case IconStyle.sharp: return '-sharp';
    }
    return '';
```

```
    switch (this) {
      case IconStyle.regular: return '';
      case IconStyle.outlined: return '-outlined';
      case IconStyle.rounded: return '-round';
      case IconStyle.sharp: return '-sharp';
      default: return '';
    }
```